README.md

```diff
@@ -1 +1,197 @@
+listener.listen(() => {
+  console.log('listening')
+
+  pull(
+    circuit.dial(mh1),
+    pull.log,
+    pull.onEnd(() => {
+      circuit.close()
+    })
+  )
+})
+```
+
+Outputs:
+
+```sh
+listening
+new connection opened
+hello
+```
+
+#### Create `relay`
+
+```js
+const Relay = require('libp2p-circuit').Realy
+
+const relay = new Relay(options)
+
+relay.mount(swarmInstance) // start relaying traffic
+```
+
+### This module uses `pull-streams`
+
+We expose a streaming interface based on `pull-streams`, rather then on the Node.js core streams implementation (aka Node.js streams). `pull-streams` offers us a better mechanism for error handling and flow control guarantees. If you would like to know more about why we did this, see the discussion at this [issue](https://github.com/ipfs/js-ipfs/issues/362).
+
+You can learn more about pull-streams at:
+
+- [The history of Node.js streams, nodebp April 2014](https://www.youtube.com/watch?v=g5ewQEuXjsQ)
+- [The history of streams, 2016](http://dominictarr.com/post/145135293917/history-of-streams)
+- [pull-streams, the simple streaming primitive](http://dominictarr.com/post/149248845122/pull-streams-pull-streams-are-a-very-simple)
+- [pull-streams documentation](https://pull-stream.github.io/)
+
+#### Converting `pull-streams` to Node.js Streams
+
+If you are a Node.js streams user, you can convert a pull-stream to a Node.js stream using the module [`pull-stream-to-stream`](https://github.com/dominictarr/pull-stream-to-stream), giving you an instance of a Node.js stream that is linked to the pull-stream. For example:
+
+```js
+const pullToStream = require('pull-stream-to-stream')
+
+const nodeStreamInstance = pullToStream(pullStreamInstance)
+// nodeStreamInstance is an instance of a Node.js Stream
+```
+
+To learn more about this utility, visit https://pull-stream.github.io/#pull-stream-to-stream.
+
+## API
+
+[![](https://raw.githubusercontent.com/libp2p/interface-transport/master/img/badge.png)](https://github.com/libp2p/interface-transport)
+
+`libp2p-circuit` accepts Circuit addresses for both IPFS and non IPFS encapsulated addresses, i.e:
+
+`/p2p-circuit/ip4/127.0.0.1/tcp/4001/ipfs/QmHash`
+
+Both for dialing and listening.
+
+### Implementation rational
+
+This module is not a transport, however it implements `interface-transport` interface in order to allow circuit to be plugged with `libp2p-swarm`. The rational behind it is that, `libp2p-circuit` has a dial and listen flow, which fits nicely with other transports, moreover, it requires the _raw_ connection to be encrypted and muxed just as a regular transport's connection does. All in all, `interface-transport` ended up being the correct level of abstraction for circuit, as well as allowed us to reuse existing integration points in `libp2p-swarm` and `libp2p` without adding any ad-hoc logic. All parts of `interface-transport` are used, including `.getAddr` which returns a list of `/p2p-circuit` addresses that circuit is currently listening.
```

Comment:
It looks like a transport, it quacks like a transport but it's not a transport?

Reply:
hrm... yeah :)

Reply:
So, it doesn't implement any transport protocol itself, it only hooks into the flow and delegates the communication to the underlying transports through swarm... There was some back and forth on why the interface was chousen, hence the explanation. This might need a bit more clarification as to why it is not a real transport.